CM.Middleware/ApiRequestAuthenticatorAttribute.cs

```diff
@@ -0,0 +1,47 @@
+﻿
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace CM.Middleware
+{
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = true, Inherited = true)]
```

Comment:
Good that you have tried out an attribute. Can you also come up with a middleware for the same implementation and throw the UnAuthorizedAccess exception and handle it with the GlobalErrorHandler middleware?